src/daft-io/src/lib.rs

```diff
@@ -133,6 +133,18 @@ impl From<Error> for DaftError {
             ReadTimeout { .. } => DaftError::ReadTimeout(err.into()),
             UnableToReadBytes { .. } => DaftError::ByteStreamError(err.into()),
             SocketError { .. } => DaftError::SocketError(err.into()),
+            // We have to repeat everything above for the case we have an Arc since we can't move the error.
+            CachedError { ref source } => match source.as_ref() {
+                NotFound { path, source: _ } => DaftError::FileNotFound {
+                    path: path.clone(),
+                    source: err.into(),
+                },
+                ConnectTimeout { .. } => DaftError::ConnectTimeout(err.into()),
+                ReadTimeout { .. } => DaftError::ReadTimeout(err.into()),
+                UnableToReadBytes { .. } => DaftError::ByteStreamError(err.into()),
+                SocketError { .. } => DaftError::SocketError(err.into()),
+                _ => DaftError::External(err.into()),
+            },
```

Comment:
@samster25 Couldn't we do a recursive match, e.g.
```suggestion
            CachedError { ref source } => source.as_ref().clone().into(),
            },
```

Reply:
I gave that a shot and the issue is that we cant clone the error. I tried implementing clone for it, but the Box<dyn Errors> are not clonable. Also another way of looking at it is that if the error was clonable we wouldn't need `CachedError`!